addon/mixins/adapter-fetch.js

```diff
@@ -0,0 +1,189 @@
+import Ember from 'ember';
+import fetch from 'fetch';
+
+const {
+  assign,
+  merge,
+  RSVP
+} = Ember;
+
+/**
+ * Helper function that turns the data/body of a request into a query param string.
+ * This is directly copied from jQuery.param.
+ * @param {Object} queryParamsObject
+ * @returns {String}
+ */
+export function serialiazeQueryParams(queryParamsObject) {
+  var s = [], rbracket = /\[\]$/;
+  function isArray(obj) {
+    return Object.prototype.toString.call(obj) === '[object Array]';
+  }
+  function add(k, v) {
```

Comment:
lets not redefine this method on each invocation of `serialiazeQueryParams`, instead hoist it up.